index.d.ts

```diff
@@ -3001,11 +3001,11 @@ declare module sequelize {
   }
 
   export interface OrOperator {
-    $or: WhereOperators | WhereAttributeHash | Array<WhereOperators | WhereAttributeHash>;
+    $or: WhereOperators | WhereAttributeHash | Array<Array<number> | WhereOperators | WhereAttributeHash>;
```

Comment:
Probably true, but tbh, this feels kinda out of scope for this PR as its primary goal is to restore compatibility with the current version of TS.
I'm open to taking a look at extending/fixing this further but I'd like to unblock my upgrade to TS 2.4 first.